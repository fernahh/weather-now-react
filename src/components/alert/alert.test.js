import React from 'react'
import { shallow } from 'enzyme'
import Alert from './alert'

describe('<Alert />', () => {
  const action = jest.fn()
  const wrapper = shallow(<Alert message="Foobar" retryAction={action} />)
  
  it('render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  
  it('execute retry action', () => {
    wrapper.find('Button').simulate('click')
    expect(action).toBeCalled()
  })
})
