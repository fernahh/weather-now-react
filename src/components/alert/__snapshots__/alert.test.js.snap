// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Alert
    message="Foobar"
    retryAction={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="alert__message"
>
          Foobar
</p>,
        <Button
          onClick={[Function]}
          type="button"
>
          Try Again
</Button>,
      ],
      "className": "alert",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Foobar",
          "className": "alert__message",
        },
        "ref": null,
        "rendered": "Foobar",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Try Again",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Try Again",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="alert__message"
>
            Foobar
</p>,
          <Button
            onClick={[Function]}
            type="button"
>
            Try Again
</Button>,
        ],
        "className": "alert",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Foobar",
            "className": "alert__message",
          },
          "ref": null,
          "rendered": "Foobar",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Try Again",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Try Again",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
